File: Sheet #2 (Loops) /C++/C-Even, Odd, Positive and Negative.cpp
#include <iostream>
using namespace std;
int main()
{
    int number_of_vales, i, even = 0, odd = 0, positive = 0, negative = 0;
    long long num;

    cin >> number_of_vales;

    for (i = 0 ; i < number_of_vales ; i++)
    {
        cin >> num;
        if (num > 0)
            positive++;
        if (num < 0)
            negative++;
        if (num % 2 == 0)
            even++;
        else
            odd++;
    }

    cout << "Even: " << even << endl;
    cout << "Odd: " << odd << endl;
    cout << "Positive: " << positive << endl;
    cout << "Negative: " << negative << endl;

    return (0);
}

File: Sheet #2 (Loops) /python/C-Even, Odd, Positive and Negative.py
#!/usr/bin/python3

test_case = int(input(""))

even = 0
odd = 0
positive = 0
negative = 0

input = input("")
input_list = input.split(" ")
input_list = [int(x) for x in input_list]

for test in input_list:
    if test % 2 == 0:
        even += 1
    if test % 2 != 0:
        odd += 1
    if test > 0:
        positive += 1
    if test < 0:
        negative += 1

print(f"Even: {even}")
print(f"Odd: {odd}")
print(f"Positive: {positive}")
print(f"Negative: {negative}")

